### ISSUE TITLE
sending money from a tfchain wallet with many addresses results in an error

### ISSUE THREAD

**Issue by GlenDC** [threefoldtech/jumpscale_lib#212]
```stacktrace
[<n>]: wallet2.send_money(amount=<amount>, recipient='<address>')
[Mon29 19:52] - RivineWallet.py   :195 :in.rivine.rivinewallet - INFO     - Current chain height is: <m>
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/code/github/threefoldtech/jumpscale_core/cmds/js_shell in <module>()
----> 1 wallet2.send_money(amount=<amount>, recipient='<address>')

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/RivineWallet.py in send_money(self, amount, recipient, data, locktime)
    347                                                 sign_transaction=True,
    348                                                 custom_data=data,
--> 349                                                 locktime=locktime)
    350         self._commit_transaction(transaction=transaction)
    351         return transaction

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/RivineWallet.py in _create_transaction(self, amount, recipient, minerfee, sign_transaction, custom_data, locktime)
    537
    538
--> 539         input_results, used_addresses, minerfee, remainder = self._get_inputs(amount=amount)
    540         for input_result in input_results:
    541             transaction.add_coin_input(**input_result)

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/RivineWallet.py in _get_inputs(self, amount, minerfee)
    435         if minerfee is None:
    436             minerfee = self._minerfee
--> 437         wallet_fund = int(self.current_balance.unlocked_balance * HASTINGS_TFT_VALUE)
    438         required_funds = amount + minerfee
    439         if required_funds > wallet_fund:

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/RivineWallet.py in current_balance(self)
     99         Retrieves current wallet balance
    100         """
--> 101         self._check_balance()
    102         wb = WalletBalance()
    103         for id, output in self._unspent_coins_outputs.items():

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/RivineWallet.py in _check_balance(self)
    230                 self._collect_transaction_outputs(current_height=current_chain_height,
    231                                                   address=address,
--> 232                                                   transactions=transactions)
    233         # Add multisig addresses
    234         for address in self._multisig_wallets:

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/RivineWallet.py in _collect_transaction_outputs(self, current_height, address, transactions)
    286         self._unconfirmed_unspent_multisig_outputs = {}
    287         self._unconfirmed_locked_multisig_outputs = {}
--> 288         txn_outputs = utils.collect_transaction_outputs(current_height, address, confirmed_tx, unconfirmed_txs)
    289         self._unspent_coins_outputs.update(txn_outputs['unlocked'])
    290         self._locked_coin_outputs.update(txn_outputs['locked'])

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/utils.py in collect_transaction_outputs(current_height, address, transactions, unconfirmed_txs)
    370         if coinoutputs:
    371             for index, utxo in enumerate(coinoutputs):
--> 372                 condition_ulh = get_unlockhash_from_output(output=utxo, address=address, current_height=current_height)
    373
    374                 if address in condition_ulh['locked'] or address in condition_ulh['unlocked']:

~/code/github/threefoldtech/jumpscale_lib/JumpscaleLib/clients/blockchain/rivine/utils.py in get_unlockhash_from_output(output, address, current_height)
    291                 # locktime represent timestamp
    292                 current_time = time.time()
--> 293                 if output['condition']['data']['condition']['type'] == 1:
    294                     # locktime should be checked against the current time
    295                     if current_time > locktime:

KeyError: 'type'
```

**Comment by robvanmieghem** [comment]
I suspect it has nothing to do with the amount of addresses but an old-style output was selected  which is not handled properly.
@LeeSmet Can  you add some logging to print the transaction so we at least know what is going on?

**Comment by LeeSmet** [comment]
It seems to be more a problem with a timelocked output, will double check the encoding of such outputs

**Comment by LeeSmet** [comment]
try except has been added around the offending section, printing the output and asking to post it here

**Comment by GlenDC** [comment]
`{'value': '225000000000', 'condition': {'type': 3, 'data': {'locktime': 1753048800, 'condition': {}}}}`

Seems that the JS client does not handle the Nil condition in any way. Will add support for it now :) Thanks Lee for the print trace, helps a lot, will remove that one as well, and provide debug info.

**Comment by GlenDC** [comment]
Closed by efc9899.
